packages/flutter/test/painting/text_painter_test.dart

```diff
@@ -145,20 +145,77 @@ void main() {
       caretOffset = painter.getOffsetForCaret(ui.TextPosition(offset: text.length), ui.Rect.zero);
       expect(caretOffset.dx, painter.width);
 
+      /// Verify the handling of spaces by SkParagraph and TextPainter
+      ///
+      /// Test characters that are in the Unicode-Zs category but are not treated as whitespace characters by SkParagraph.
+      /// The following character codes are intentionally excluded from the test target.
+      ///   * '\u{00A0}' (no-break space)
+      ///   * '\u{2007}' (figure space)
+      ///   * '\u{202F}' (narrow no-break space)
+      void verifyCharacterIsConsideredTrailingSpace(String character) {
+        final String reason = 'character: ${character.codeUnitAt(0).toRadixString(16)}';
+
+        text = 'A$character';
+        checkCaretOffsetsLtr(text);
+        painter.text = TextSpan(text: text);
+        painter.layout();
+        caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 0), ui.Rect.zero);
+        expect(caretOffset.dx, 0.0, reason: reason);
+        caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 1), ui.Rect.zero);
+        expect(caretOffset.dx, 14.0, reason: reason);
+        caretOffset = painter.getOffsetForCaret(ui.TextPosition(offset: text.length), ui.Rect.zero);
+        expect(caretOffset.dx, painter.width, reason: reason);
+
+        painter.layout(maxWidth: 14.0);
+        final List<ui.LineMetrics> lines = painter.computeLineMetrics();
+        expect(lines.length, 1, reason: reason);
+        expect(lines.first.width, 14.0, reason: reason);
+      }
+
+      // Test with trailing space
```

Comment:
Nit: I think there should be a period at the end of each of these comments here and below.